[FEATURE] - replace elasticsearch with opensearch (#67)
sighupio/fury-kubernetes-logging: katalog/opensearch-triple/README.md

# OpenSearch Triple

<!-- <KFD-DOCS> -->

OpenSearch is an open-source distributed search and analytics engine used for
log analytics. This package deploys a three-node OpenSearch cluster on
Kubernetes.

`opensearch-triple` is a high availability setup of OpenSearch, that sets
up a 3-node cluster of `OpenSearch` for a robust and reliable setup.

## Requirements

- Kubernetes >= `1.20.0`
- Kustomize >= `v3.3.X`
- [prometheus-operator][prometheus-operator]

> Prometheus Operator is necessary since we configure a `ServiceMonitor` to make
> some metrics available from `opensearch` on prometheus. Please refer,

> > [`opensearch-single](../opensearch-single/README.md#alerts) to read

Reviewer: ```suggestion
> [opensearch-single](../opensearch-single/README.md#alerts) to read
```

> about the available Prometheus rules.

## Image repository and tag

> * Elasticsearch image: `elasticsearch/elasticsearch:7.16.2`
> * Elasticsearch repo: [Elasticsearch on Github][es-gh]
> * Elasticsearch documentation: [Elasticsearch Homepage][es-doc]

Reviewer: ```suggestion
* OpenSearch image: `opensearchproject/opensearch:1.2.4`
* OpenSearch repo: [OpenSearch on Github][opensearch-gh]
* OpenSearch documentation: [OpenSearch Homepage][opensearch-doc]
```

## Configuration

> Fury distribution OpenSearch Single is deployed with the following configuration:

Reviewer: ```suggestion
OpenSearch Triple is deployed with the following configuration:
```

- OpenSearch cluster with `3` nodes
- Listens on port `9200` for client connections
- Listens on port `9300` for node-to-node connections
- Resource limits are `2000m` for CPU and `4G` for memory
- Requires `30Gi` storage

> - Only a single OpenSearch node can be deployed on each node of the cluster

Reviewer: ```suggestion
- OpenSearch nodes running in different Kubernetes nodes
```

> - Prometheus exporter to expose Opensearch metrics

Reviewer: ```suggestion
- Prometheus exporter to expose OpenSearch metrics
```
